src/navigators/createMaterialBottomTabNavigator.js

```diff
@@ -18,19 +18,32 @@ class BottomNavigationView extends React.Component<Props> {
     return options.tabBarColor;
   };
 
+  _isVisible = ({ route }) => {
+    const { descriptors } = this.props;
+    const descriptor = descriptors[route.key];
+    const options = descriptor.options;
+
+    return options.tabBarVisible;
+  };
+
   render() {
     const {
       activeTintColor,
       inactiveTintColor,
       navigation,
       // eslint-disable-next-line no-unused-vars
       descriptors,
+      barStyle
       ...rest
     } = this.props;
+    if (!this._isVisible()) {
+    	barStyle['display'] = 'none';
```

Comment:
`barStyle` is not guaranteed to be an object (it can be an array of styles, a style id which is a number, or a falsy value such as `null`) so we can't do it. We'd also want the `tabBarVisible` option to compose with the `barStyle` option properly.

So we'll need to do something like this:

```js
const isVisible = this._isVisible();
const extraStyle = typeof isVisible === 'boolean' ? { display: isVisible ? 'auto' : 'none' } : null;

...

barStyle={[barStyle, extraStyle]}
```